src/test/scala/io/qbeast/spark/delta/QbeastSnapshotTest.scala

```diff
@@ -249,16 +238,11 @@ class QbeastSnapshotTest extends QbeastIntegrationTestSpec {
 
       spark.read.format("qbeast").load(tmpDir).count.toInt shouldBe 100
 
-      val deltaLog = DeltaLog.forTable(spark, tmpDir)
-      val qbeastSnapshot = DeltaQbeastSnapshot(deltaLog.update())
+      val qbeastSnapshot = getQbeastSnapshot(tmpDir)
 
-      val lastRev = qbeastSnapshot.loadLatestRevision
-      val files = qbeastSnapshot.loadAllRevisions
-        .map(rev => qbeastSnapshot.loadRevisionFiles(rev.revisionID))
-        .foldLeft(spark.emptyDataset[AddFile])(_ union _)
-      import spark.implicits._
-      val filesToOptimize: Dataset[IndexFile] =
-        files.map(IndexFiles.fromAddFile(lastRev.columnTransformers.size))
+      val filesToOptimize = qbeastSnapshot.loadAllRevisions
```

Comment:
should we move the `filesToOptimize` code into a method? Make it more testable this way

Reply:
For now, I would keep it like this since it is only called twice. I think we can consider, in the future, doing a pass through all the tests in a separate issue to see if we can improve them in some way.